packages/forms-web-app/src/validators/common/schemas/check-document-uploaded-schema.js

```diff
@@ -0,0 +1,38 @@
+const { findTargetValueInJSON } = require('../../../lib/find-target-value-in-json');
+
+const schema = (path, documentType, submissionType) => ({
+	[path]: {
+		custom: {
+			options: async (value, { req }) => {
+				if (!req.files) {
+					let uploadedFiles;
+
```

Comment:
I think we could simplify the selection of the uploaded files somewhat (again tried in github not an IDE :) )
```js
const uploadedFiles = req.session[submissionType]?.[documentType]?.uploadedFiles;
```

Reply:
this would work for final comments, where the uploaded documents appear in the location you indicate - i.e. `req.session.finalComment.supportingDocuments.uploadedFile`, but in an appeal object, uploaded documents have an extra layer of nesting , i.e. `req.session.appeal.planningApplicationDocumentsSection.plansDrawingsSupportingDocuments.uploadedFiles`, the `findTargetValueInJSON` function is a way around this.

alternatively, for appeals, I believe that [sectionName] is part of the req for specific pages, so this could be used - i.e. if `req.session[submissionType]?.[documentType]?.uploadedFiles` is undefined, try ``req.session[submissionType]?.[sectionName]?.[documentType]?.uploadedFiles` - but then we're becoming quite tied to the structure of the objects?